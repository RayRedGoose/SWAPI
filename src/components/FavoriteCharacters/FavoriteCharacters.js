import React from 'react';
import CharacterCard from '../CharacterCard/CharacterCard';
import './FavoriteCharacters.scss';
import FavoriteFilter from '../FavoriteFilter/FavoriteFilter'

const FavoriteCharacters = ({favorite}) => {
  // const putCards = favorite.map((character,ind) => (
  //   <CharacterCard
  //     key={ind}
  //     name={character.name}
  //     species={character.species}
  //     planet={character.planet}
  //     population={character.population}
  //     films={character.films}
  //   />
  // ))

  return (
    <section className="favorite-box">
      <FavoriteFilter />
      {
        //putCards
      }
    </section>
  )
}

export default FavoriteCharacters
