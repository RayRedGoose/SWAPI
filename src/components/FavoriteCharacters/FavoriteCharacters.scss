.favorite-box {
  margin-bottom: 20px;
}
