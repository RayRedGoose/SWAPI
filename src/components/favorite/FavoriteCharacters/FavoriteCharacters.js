import React from 'react'
import CharacterCard from '../../CharacterCard/CharacterCard'
import NoFavoritesPage from '../NoFavoritesPage/NoFavoritesPage'
import PropTypes from 'prop-types'
import './FavoriteCharacters.scss'

const FavoriteCharacters = ({favorites, removeFavorite}) => {
  const putCards = favorites.length? favorites.map((character,ind) => (
    <CharacterCard
      key={ind}
      character={character}
      isFavorite={true}
      removeFavorite={removeFavorite}
    />
  )) : <NoFavoritesPage />

  return (
    <section className="favorite-box">
      {putCards}
    </section>
  )
}

export default FavoriteCharacters

FavoriteCharacters.propTypes = {
  favorites: PropTypes.arrayOf(PropTypes.object),
  removeFavorite: PropTypes.func.isRequired
}
