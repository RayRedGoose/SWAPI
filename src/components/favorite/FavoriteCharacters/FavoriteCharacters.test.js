import React from 'react'
import characterResults from '../../../mockData/fakeCharacters.js'
import { shallow } from 'enzyme'
import FavoriteCharacters from './FavoriteCharacters'

describe('FavoriteCharacters', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(
      <FavoriteCharacters
        favorites={[characterResults[1]]}
        removeFavorite={jest.fn()}
      />)
  })

  it('should render FavoriteCharacters with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
