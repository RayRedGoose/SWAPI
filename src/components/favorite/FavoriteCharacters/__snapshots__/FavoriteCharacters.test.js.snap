// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteCharacters should render FavoriteCharacters with all data passed in correctly 1`] = `
<section
  className="favorite-box"
>
  <CharacterCard
    character={
      Object {
        "films": Array [
          "die hard",
        ],
        "isFavorite": true,
        "name": "Leia",
        "planet": "Alderaan",
        "population": 230000,
        "species": "human",
      }
    }
    isFavorite={true}
    key="0"
    removeFavorite={[MockFunction]}
  />
</section>
`;
